File: Time travel.r
TimeMachine <- function() {
  current_time <- 0
  
  travel <- function(years) {
    current_time <<- current_time + years
  }
  
  get_current_time <- function() {
    return(current_time)
  }
  
  return(list(travel = travel, get_current_time = get_current_time))
}

demonstrate_time_travel <- function() {
  time_machine <- TimeMachine()
  
  cat("Welcome to the Time Machine!\n")
  cat("Current time:", time_machine$get_current_time(), "\n")
  
  # Travel forward in time
  years_forward <- as.integer(readline("Enter the number of years to travel forward: "))
  time_machine$travel(years_forward)
  cat("Traveled forward in time by", years_forward, "years.\n")
  cat("Current time:", time_machine$get_current_time(), "\n")
  
  # Travel backward in time
  years_backward <- as.integer(readline("Enter the number of years to travel backward: "))
  time_machine$travel(-years_backward)
  cat("Traveled backward in time by", years_backward, "years.\n")
  cat("Current time:", time_machine$get_current_time(), "\n")
}

demonstrate_time_travel()

File: Time travel.rs
struct TimeMachine {
    current_time: i32,
}

impl TimeMachine {
    fn new() -> TimeMachine {
        TimeMachine { current_time: 0 }
    }

    fn travel(&mut self, years: i32) {
        self.current_time += years;
    }

    fn get_current_time(&self) -> i32 {
        self.current_time
    }
}

fn demonstrate_time_travel() {
    let mut time_machine = TimeMachine::new();

    println!("Welcome to the Time Machine!");
    println!("Current time: {}", time_machine.get_current_time());

    // Travel forward in time
    println!("Enter the number of years to travel forward: ");
    let mut input = String::new();
    std::io::stdin().read_line(&mut input).expect("Failed to read input");
    let years_forward: i32 = input.trim().parse().expect("Invalid input");
    time_machine.travel(years_forward);
    println!("Traveled forward in time by {} years.", years_forward);
    println!("Current time: {}", time_machine.get_current_time());

    // Travel backward in time
    println!("Enter the number of years to travel backward: ");
    input.clear();
    std::io::stdin().read_line(&mut input).expect("Failed to read input");
    let years_backward: i32 = input.trim().parse().expect("Invalid input");
    time_machine.travel(-years_backward);
    println!("Traveled backward in time by {} years.", years_backward);
    println!("Current time: {}", time_machine.get_current_time());
}

fn main() {
    demonstrate_time_travel();
}